src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -153,6 +153,12 @@ static void HandleNodeBrowse(void * context, DnssdService * services, size_t ser
 
     if (error != CHIP_NO_ERROR)
     {
+        // We don't have access to the ResolverProxy here to clear out its
+        // mDiscoveryContext.  The underlying implementation of
+        // ChipDnssdStopBrowse needs to handle a possibly-stale reference
+        // anyway, though, so it's OK.
```

Comment:
The interface implies that this isn't completely feasible.  Yes, if `mDiscoveryContext` is passed to `ChipDnssdStopBrowse` and no matching intptr_t is found, it is obvious that this should be a no-op.

But if an intptr_t is passed for a stale operation, but this happens to match the intptr_t for a new, unrelated browse, the underlying implementation will presumably cancel the something that it shouldn't.

TODO is fine.  But I think this is less OK than this comment implies.  This can break and do the wrong thing.

Reply:
That's fair.  I will update the comment to make the implications clear, and in a followup see if I can get access to the ResolverProxy in this code usefully.